In [13]:
import numpy as np
from scipy import stats
from statsmodels.stats.multicomp import pairwise_tukeyhsd
import statistics

# Problemas de prueba de hipótesis

## Ejercicio 1

Usted es un nutricionista que investiga dos tipos diferentes de dietas para ver si existe una diferencia significativa en la pérdida de peso después de un mes. Eliges dos grupos aleatorios de personas; un grupo sigue la primera dieta y el otro sigue la segunda. Al final del mes, se registra la pérdida de peso (en kg) de cada persona:

| Diet 1 | Diet 2 |
|:-------|:-------|
| 2.0 | 3.0 |
| 2.5 | 3.2 |
| 3.0 | 3.1 |
| 2.8 | 2.9 |
| 2.3 | 2.8 |
| 2.7 | 3.0 |
| 2.5 | 3.2 |

Con estos datos se busca responder a la siguiente pregunta: ¿Existe una diferencia significativa en la pérdida de peso promedio entre las personas que siguieron la primera dieta y las que siguieron la segunda dieta?

Para sacar conclusiones, siga los siguientes puntos:

-Enunciar la hipótesis: hipótesis nula y alternativa.
-Realizar la prueba para comprobar la hipótesis. Puede utilizar la prueba t de Student.
-Analizar las conclusiones.

In [8]:
# hipotesis nula H0= no hay diferencia significativa en la perdida de peso promedio entre las dos dietas
# Hipotesis alternativa H1 = hay difenrecia significativa en la perdidad de peso promedio entre las dos dietas
dieta1 = np.array([2.0, 2.5, 3.0, 2.8, 2.3, 2.7, 2.5])
dieta2 = np.array([3.0, 3.2, 3.1, 2.9, 2.8, 3.0, 3.2])

t_stat, p_value = stats.ttest_ind(dieta1,dieta2)
print(f"Estadístico t:, {t_stat}")
print(f"Valor P: {p_value}")

Estadístico t:, -3.5383407969933938
Valor P: 0.004083270191713912


Tenemos que el P_valor es < 0.05 por lo cual se rechaza la hipotesis nula y se acepta la hipotesis alternativa (hay difenrecia significativa en la perdidad de peso promedio entre las dos dietas)

## ANOVA

**ANOVA** (*Analysis of Variance - Análisis de variación*) Es una técnica estadística utilizada para comparar las medidas de dos o más grupos. La idea detrás de ANOVA es descomponer la variabilidad total de los datos en dos componentes: variabilidad entre grupos y variabilidad dentro de los grupos:

- **Variabilidad entre grupos**: Esta variabilidad se refiere a las diferencias entre las medias del grupo. Si esta variabilidad es considerablemente mayor que la variabilidad dentro del grupo, podría ser una indicación de que al menos una de las medias del grupo es diferente.
- **Variabilidad dentro del grupo**: Esta variabilidad se refiere a la dispersión de los datos dentro de cada grupo. Si todos los grupos tienen una variabilidad similar, entonces cualquier diferencia notable en las medias de los grupos podría considerarse significativa.

Las hipótesis en ANOVA suelen incluir:

- **Hipótesis nula** ($H₀$): Las medias de todos los grupos son iguales.
- **Hipótesis alternativa** ($H₁$): Al menos una de las medias del grupo es diferente.

Si el resultado de la prueba ANOVA es significativo (por ejemplo, un valor p inferior a un umbral como 0,05), esto sugiere que al menos la media de un grupo es diferente.

## Ejercicio 2

Un agricultor decide probar tres tipos diferentes de fertilizantes para determinar si alguno es superior en términos de producción de maíz. El agricultor planta maíz en 15 parcelas idénticas y utiliza los tres fertilizantes (5 parcelas para cada tipo). Al final de la temporada mide el rendimiento de maíz (en kg) de cada parcela, con el siguiente resultado:

| Fertilizante 1 | Fertilizante 2 | Fertilizante 3 |
|:-------------|:-------------|:-------------|
| 20 | 22 | 24 |
| 21 | 21 | 23 |
| 20 | 23 | 22 |
| 19 | 22 | 23 |
| 20 | 21 | 24 |

Con estos datos busca responder la siguiente pregunta: ¿Existe una diferencia significativa en el rendimiento promedio del maíz entre los tres tipos de fertilizantes?

Para ayudarte, sigue los siguientes puntos:

-Enunciar la hipótesis: hipótesis nula y alternativa.
-Realizar la prueba ANOVA.
-Analizar las conclusiones.
-Si un fertilizante es mejor que otro, ¿cómo podemos saberlo?

In [ ]:
# Hipotesis Nula H0 = la media de todos los grupos son iguales 
# Hipotesus Alternativa H1 = la media de al menos 1 grupo es diferente
fertilizante_1 = [20, 21, 20, 19, 20]
fertilizante_2 = [22, 21, 23, 22, 21]
fertilizante_3 = [24, 23, 22, 23, 24]

# ANOVA test
f_value, p_value = stats.f_oneway(fertilizante_1, fertilizante_2, fertilizante_3)

print(f"Valor F: {f_value}")
print(f"Valor P: {p_value}")

Valor F: 20.31578947368421
Valor P: 0.000140478247931904


Tenemos que el P_valor es < 0.05 por lo cual se rechaza la hipotesis nula y se acepta la hipotesis alternativa (la media de al menos 1 grupo es diferente)

In [11]:
data = np.concatenate([fertilizante_1, fertilizante_2, fertilizante_3])
labels = ["F1"] * 5 + ["F2"] * 5 + ["F3"] * 5

# Tukey test
result = pairwise_tukeyhsd(data, labels, alpha = 0.05)
print(result)

Multiple Comparison of Means - Tukey HSD, FWER=0.05
group1 group2 meandiff p-adj  lower  upper  reject
--------------------------------------------------
    F1     F2      1.8 0.0099 0.4572 3.1428   True
    F1     F3      3.2 0.0001 1.8572 4.5428   True
    F2     F3      1.4 0.0409 0.0572 2.7428   True
--------------------------------------------------


Con el metodo tukey podemos ver que el fetilizante 3 es el que posee una media mas elevanda frente al 1 que seria 3.2 por encima y frente al 2 que seria 1.4 por encima, por lo que este seria el mejor (Fertilizante 3).
Pd, justamente la interpretacion que iba a hacer era sacando la media de cada fertilizante y comparando, pero me quedo con este metodo que propone la solucion ya que me parecio mucho mas intereseante. porque ademas de que compara la difenrecia que hay entre las medias, tambien mide el P_ajustado entre la comparacion de los grupos, es decir si hay difrencia significativa (p-adj<0.05) o si no la hay (p-adj>0.05). En este caso podemos ver que hay una diferencia significativa entre todos.